### DISCUSSION TITLE
Is there a way to listen to a form reset?

### DISCUSSION THREAD

**Discussion by bellizio** [react-hook-form/react-hook-form#2993] [Q&A] [answered]
For example, say I have a nested child component of a form and I want to perform a side effect whenever the form is reset via the `reset()` method. 

Is there a way to listen to the reset event? I am using `isSubmitSuccessful` from the form state elsewhere, so I was thinking something similar to that, `submitCount`, or some other formState prop.

**Comment by bluebill1049** [comment, marked as answer]
`dirty`?

**Reply by bellizio** [reply]
That seems to do the trick.

Ended up with something like this:
```
const { formState } = useFormContext();
const { isSubmitted, isDirty } = formState;

useEffect(() => {
  if (!isSubmitted && !isDirty) {
    // do something
  }
}, [isDirty]);
```

Thanks for the reply and suggestion!

**Reply by bellizio** [reply]
So this actually did not end up working as I expected it would.

The reason being is that I ran into an issue where if an input field is cleared after having a value entered into it, then `isDirty` is reset to `false`. As a result, the branch of code in the if block is executed.

Shouldn't `isDirty` remain as `true` in this scenario?

**Reply by bellizio** [reply]
Ok, it is working as expected after updating to v6.8.6, so I'm assuming the issue described above was fixed in a later version than what I was on.

Re-marking your comment as the answer.

**Reply by Haraldson** [reply]
It’s unclear how anything was solved by upgrading. What is it that started working as expected, @bellizio?

I too want to listen in on the reset event, and am not sure how the combination of those two states would help me achieve that…

**Reply by bellizio** [reply]
@Haraldson it's been a while since I solved this but looking back on it I think it was related to [this issue](https://github.com/react-hook-form/react-hook-form/pull/2948).

If it helps, this is basically the final code snippet of what I ended up with for my use case:
```
useEffect(() => {
        if (isFirstRender.current) {
            isFirstRender.current = false;
            return;
        }
        // this is a way to listen for the form reset() function call
        if (!isSubmitted && !isDirty) {
            // clear value when form is reset
            setStateValue(0);
            callbackFn(0);
        }
    }, [isDirty]);
```

**Reply by Haraldson** [reply]
@bellizio Thanks for taking the time anyway! I guess this logic kind of makes sense, but also relyes heavily on behavior that could change. @bluebill1049 Wouldn’t it be an idea to expose an `isReset` boolean which is `true` after resetting the form, until any further interaction/dirtying takes place?

**Reply by bluebill1049** [reply]
Do you want to send a feature request on this? We can ping on the feature board and see if users have enough demand on this.

**Reply by Haraldson** [reply]
I could possibly write up a better feature request/spec/use case for it, but I just don’t have the energy and headspace these days to get to a level of insight into RHF’s code base where I could meaningfully contribute code.

**Comment by mp3por** [comment]
Is there a better way now ?
